src/protocol/boolean/prefix_or.rs

```diff
@@ -21,35 +21,47 @@ use super::BitOpStep;
 /// 5.2 Prefix-Or
 /// "Unconditionally Secure Constant-Rounds Multi-party Computation for Equality, Comparison, Bits, and Exponentiation"
 /// I. Damgård et al.
-pub struct PrefixOr<'a, B: BinaryField> {
-    input: &'a [Replicated<B>],
+pub struct PrefixOr<'a, F: Field> {
+    input: &'a [Replicated<F>],
 }
 
-impl<'a, B: BinaryField> PrefixOr<'a, B> {
+impl<'a, F: Field> PrefixOr<'a, F> {
     #[allow(dead_code)]
-    pub fn new(input: &'a [Replicated<B>]) -> Self {
+    pub fn new(input: &'a [Replicated<F>]) -> Self {
         Self { input }
     }
 
-    /// Securely computes `[a] | [b] where a, b ∈ {0, 1}`
-    /// OR can be computed as: `[a] ^ [b] ^ ([a] & [b])`
+    /// Securely computes `[a] | [b] where a, b ∈ {0, 1} ⊆ F_p`
+    ///
+    /// * OR can be computed as: `[a] ^ [b] ^ MULT([a], [b])`
+    /// * XOR([a], [b]) is: `[a] + [b] - 2([a] * [b])`
+    ///
+    /// Therefore,
+    ///
+    /// let [c] = [a] ^ [b]
+    /// [c] + [ab] - 2([c] * [ab])
     async fn bit_or(
-        a: Replicated<B>,
-        b: Replicated<B>,
-        ctx: ProtocolContext<'_, Replicated<B>, B>,
+        a: Replicated<F>,
+        b: Replicated<F>,
+        ctx: ProtocolContext<'_, Replicated<F>, F>,
         record_id: RecordId,
-    ) -> Result<Replicated<B>, BoxError> {
-        let a_and_b = ctx.multiply(record_id, a, b).await?;
-        Ok(a + b + a_and_b)
+    ) -> Result<Replicated<F>, BoxError> {
+        let ab = ctx.narrow(&Step::AMultB).multiply(record_id, a, b).await?;
+        let c = a + b - (ab * F::from(2));
+        let cab = ctx
+            .narrow(&Step::ABMultC)
+            .multiply(record_id, c, ab)
+            .await?;
+        Ok(c + ab - (cab * F::from(2)))
```

Comment:
This seems like an inefficient way of computing $OR([a], [b])$. A more efficient way immediately comes to mind:

$1- MULT([1-a], [1-b])$

But ignoring that for a moment, the paper says: "Note that we can compute an Or with unbounded fan-in:

$[x]_p$ <- $\bigvee [x_j]_p$ using section 5.1 as this is a symmetric function. 

I'm not seeing this in the code right now.